File: Abubakr Khanooni/Arrays/NumberOfGoodPairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> mp;

        for (auto it : nums) {
            mp[it]++;
        }
        
        int count = 0;

        for (auto it : mp) {
            int occurrences = it.second;
            if (occurrences > 1) {
                count += (occurrences * (occurrences - 1)) / 2;
            }
        }
        
        return count;
    }
};

File: Khushboo Singh/Array/Number of Good Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int i,j,c=0;
        for(i=0;i<nums.length;i++){
            for(j=i;j<nums.length;j++){
                if(nums[i]==nums[j] && i<j){
                    c=c+1;
                }
            }
        }
        return c;
    }
}
